commands/serve/web/components/Visualize/Visualize.jsx

```diff
@@ -97,11 +98,16 @@ export default function Visualize() {
   );
 
   useEffect(() => {
-    if (info) initiateWatch(info);
+    (async () => {
+      if (info) {
+        await initiateWatch(info);
+        setInit(true);
+      }
+    })();
   }, [info]);
 
   useEffect(() => {
-    if (waiting) return;
+    if (waiting || !initialized) return;
```

Comment:
Do we need to add `initialized` to dependency list of the useEffect?

Reply:
we do yes, good catch